lib/perl/Genome/Config/AnalysisProject/Command/UnarchiveInstrumentData.pm

```diff
@@ -0,0 +1,51 @@
+package Genome::Config::AnalysisProject::Command::UnarchiveInstrumentData;
+
+use strict;
+use warnings;
+
+use Genome;
+
+use feature qw(say);
+
+class Genome::Config::AnalysisProject::Command::UnarchiveInstrumentData {
+    is => 'Genome::Config::AnalysisProject::Command::Base',
+    has_input => [
+        volume => {
+            is => 'Genome::Disk::Volume',
+            doc => 'A location to which scratch allocation information can be stored during unarchiving',
+            is_optional => 1,
+        },
+    ],
+    doc => 'a helper that produces commands that can be run to unarchive data',
+};
+
+sub execute {
+    my $self = shift;
+    my $anp = $self->analysis_project;
+
+
+    for my $instrument_data ($anp->instrument_data) {
+        my @allocations = Genome::Disk::Allocation->get(owner_id => $instrument_data->id, owner_class_name => $instrument_data->class);
+
+        if (@allocations) {
+            for my $allocation (@allocations) {
+                if ($allocation->is_archived) {
+                    say 'genome disk allocation unarchive --analysis-project ' . $anp->id . ' ' . $allocation->id;
```

Comment:
I guess on the plus side, nobody needs to run that unarchive command just because we print it out!  But I wonder if there's a good way to distinguish between ones we care about and ones we don't.